### PR TITLE
Feat/proxy contract

### PR DESCRIPTION
**Pull request by tesol2y090**

### CODE REVIEW

File: contracts/NFTBroker.sol [hunk 1]
```diff
@@ -164,26 +167,35 @@ contract NFTBroker is Ownable, ReentrancyGuard, ERC1155Holder, INFTBroker {
     return rates[_nftAddress][_fromId][_toId];
   }
 
-  
-
-  function _addNft(address _assetAddress, uint256 _tokenId) private {
```
Comment (pisuthd) [inline contracts/NFTBroker.sol:169]: Put it back

File: contracts/NFTManager.sol [hunk 2]
```diff
@@ -21,18 +18,11 @@ import "./interfaces/IPancakeFactory.sol";
  * @dev The contract heavily depends on 3rd party modules from QuickSwap, Chainlink to running. Check out docs.tamago.finance for more details
  */
 
-contract NFTManager is ReentrancyGuard, Whitelist, INFTManager, ERC1155Holder {
```
Comment (pisuthd) [inline contracts/NFTManager.sol:24]: Missing `ERC1155Holder`

File: contracts/NFTManager.sol [hunk 3]
```diff
@@ -84,16 +72,17 @@ contract NFTManager is ReentrancyGuard, Whitelist, INFTManager, ERC1155Holder {
     // Ignore offset/discount fees when active
     bool public offsetDisabled;
     bool public discountDisabled;
+    // Multiplier for discount/redeem fees
```
Comment (pisuthd) [inline contracts/NFTManager.sol:75]: Removing multipier

File: contracts/NFTBroker.sol (hunk 1 above)
Comment (Moopz-Dev) [reply contracts/NFTBroker.sol:169]: This sub-issue is resolved in commit [937e7ac](https://github.com/tamago-finance/tamago-finance/pull/81/commits/937e7acf9055bff97515e433c2c4fab94a741c1c)

File: contracts/NFTManager.sol (hunk 2 above)
Comment (Moopz-Dev) [reply contracts/NFTManager.sol:24]: Sub-issue addressed in commit [50015c1](https://github.com/tamago-finance/tamago-finance/pull/81/commits/50015c1a58cf792183b8475ff9c78262f7a1c3dd)

File: contracts/NFTManager.sol (hunk 3 above)
Comment (Moopz-Dev) [reply contracts/NFTManager.sol]: Sub-issue addressed in commit [abd7b30](https://github.com/tamago-finance/tamago-finance/pull/81/commits/abd7b30205974583a9fe625b752d32af32949fec)

**Comment by Moopz-Dev** [comment]
As of commit [abd7b30](https://github.com/tamago-finance/tamago-finance/pull/81/commits/abd7b30205974583a9fe625b752d32af32949fec), test cases for NFTManager don't work as intended.

**Comment by Moopz-Dev** [comment]
As of commit [0c6766c](https://github.com/tamago-finance/tamago-finance/pull/81/commits/0c6766c64a321aaec2a2ce092ff7f10bbe2d1ee1),  test cases for proxy-contracts are still being worked on, their test files are intentionally commented out.

**Pull request merged**